test/backend/test_auth_providers.py

```
@@ -0,0 +1,313 @@
+        ldap3 = MagicMock()
+        ldap3.NONE = "NONE"
+        ldap3.SIMPLE = "SIMPLE"
+        ldap3.ANONYMOUS = "ANONYMOUS"
+        ldap3.AUTO_BIND_NONE = "AUTO_BIND_NONE"
+        ldap3.SUBTREE = "SUBTREE"
+
+        # Service account connection
+        svc_conn = MagicMock()
+        svc_conn.bind.return_value = bind_success
+        svc_conn.result = {"description": "success"} if bind_success else {"description": "invalidCredentials"}
+        fake_entries = [MagicMock(entry_dn=f"uid=alice,dc=example,dc=com") for _ in range(search_entries)]
```

Ruff (0.15.10)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)